customfieldnames.civix.php

```diff
@@ -346,3 +346,166 @@ function _customfieldnames_civix_civicrm_alterSettingsFolders(&$metaDataFolders
+    //validation for Group table name
+    $table_name = CRM_Utils_Array::value( 'table_name', $fields );
+    if (! $table_name ) {
+      $error_message = ts( 'Set Table Name is a required field' );
+      $form->setElementError('table_name', $error_message);
+    }
+    elseif (CRM_Core_DAO::checkTableExists($table_name)) {
+      $error_message= ts("Cannot create custom table because %1 is already exists.", array('1' => $table_name));
+      $form->setElementError('table_name', $error_message);
+    }
+    elseif (strpos($table_name, ' ')) {
+      $error_message= ts("Invalid table name. Table name should not have space in between words.");
+      $form->setElementError('table_name', $error_message);
+    }
+    //validation for Group name
+    $group_name = CRM_Utils_Array::value( 'name', $fields );
+    if (! $group_name ) {
+      $error_message = ts( 'Set Group Name is a required field' );
+      $form->setElementError('name', $error_message);
+    }
+    elseif ( ($group_id = _customfieldnames_civix_civicrm_get_group_id($group_name)) > 0 ) {
+      $error_message= ts("Cannot create custom group because %1 is already exists.", array('1' => $group_name));
+      $form->setElementError('name', $error_message);
+    }
+  }
+  // Form validation for Custom fields for a Custom Data Set
+  if($formName == 'CRM_Custom_Form_Field' && ($form->getAction() == CRM_Core_Action::ADD) ) {
+    $gid = $form->getVar( '_gid' );
+    //validation for Field Column name
+    $column_name = CRM_Utils_Array::value( 'column_name', $fields );
+    if (! $column_name ) {
+      $error_message = ts( 'Set Column Name is a required field' );
+      $form->setElementError('column_name', $error_message);
+    }
+    elseif ( ($field_id = _customfieldnames_civix_civicrm_get_custom_column_field_id($gid, $column_name)) > 0 ) {
+      $error_message= ts("Cannot create custom column because %1 is already exists.", array('1' => $column_name));
+      $form->setElementError('column_name', $error_message);
+    }
+    //validation for Field name
+    $field_name = CRM_Utils_Array::value( 'name', $fields );
+    if (! $field_name ) {
+      $error_message = ts( 'Set Field Name is a required field' );
+      $form->setElementError('name', $error_message);
+    }
+    elseif ( ($field_id = _customfieldnames_civix_civicrm_get_custom_name_field_id($gid, $field_name)) > 0 ) {
+      $error_message= ts("Cannot create custom field because %1 is already exists.", array('1' => $field_name));
+      $form->setElementError('name', $error_message);
+    }
+  }
+  return;
+}
+/**
+ * See if a CiviCRM custom field group exists
+ *
+ * @param string $group_name
+ *   custom field group name to look for, corresponds to field civicrm_custom_group.name
+ * @return integer
+ *   custom field group id if it exists, else zero
+ */
+function _customfieldnames_civix_civicrm_get_group_id($group_name){
+  $result = 0;
```

Comment:
let's use the api rather than the BAO functions in this place.

Reply:
Done. I have implemented civicrm api to get data and apply form validation.